### PR TITLE
Dashboard: Allow more than 26 queries per panel.

### LINKED ISSUE #4978: [Feature Req] Support more than 26 queries per panel
I created graph which contains distinct 30 queries (node.0.000000000001 ~ node.0.000000000030).
I expect  graph shows all metrics. But didn't show 3 metrics (node.0.000000000027 ~ node.0.000000000029). Instead shows last metrics (node.0.000000000030) 4 times.

graph
query
- What grafana version are you using?
  - Grafana v3.0.0-beta7 (2016-05-02)
- What datasource are you using?
  - graphite
- What OS are you running grafana on?
  - Ubuntu 14.04
- What did you do?
  - I created graph contains distinct 30 queries.
- What was the expected result?
  - Graph shows 30 metrics.
- What happenend instead?
  - Graph only shows 27 metrics, didn't show 3 metrics. But shows last metrics 4 times.

Below screenshot is request data of `render` API. 
render_request

I will create PR for this issue soon.

### PR DESCRIPTION
**Pull request by dfairburn**
Fixes #4978

**What this PR does / why we need it**:
This PR allows users to create more than 26 queries per panel, which was previously the upper limit.

We often encounter situations where we need many complex or simple queries for a particular panel type (e.g. Diagram Plugins).

**Which issue(s) this PR fixes**:
Open issue:
Fixes #4978

Duplicate issues:
https://github.com/grafana/grafana/issues/7783
https://github.com/grafana/grafana/issues/11129
https://github.com/grafana/grafana/issues/13934
https://github.com/grafana/grafana/issues/19056
https://github.com/grafana/grafana/issues/19200
https://github.com/grafana/grafana/issues/20184
https://github.com/grafana/grafana/issues/20868
https://github.com/grafana/grafana/issues/21802
https://github.com/grafana/grafana/issues/28104
https://github.com/grafana/grafana/issues/27957

### CODE REVIEW

**Review by tskarhed — APPROVED** [review]
Looks good! Thanks for the PR! 
Screenshot from 2021-06-10 09-16-53

**Comment by hugohaggmark** [comment]
I'm not sure this is a good idea, I think there are several reasons why we haven't fixed this yet. @torkelo is this something we want to support?

**Review by hugohaggmark — DISMISSED** [review]
I'm not sure this is a good idea, I think there are several reasons why we haven't fixed this yet. @torkelo is this something we want to support? Holding this until we've decided what to do.

**Comment by dupondje** [comment]
Well I really think we need to support this. If this breaks some other things (but on first test it does not according to @tskarhed ) we should fix those and not stay on supporting only 26 queries per panel.
This feels like saying: 32MB of RAM is perfect for everybody :)

Until 7.1.x everything worked with +26 queries per panel on Prometheus, but it broke in #27957.
And the reason it broke is just because it did not have a refId (as those are limited to 26 in total ...).
So for that we are still on 7.1.5 where everything works fine with +26 queries!

File: public/app/core/utils/query.ts [hunk 1]
```diff
@@ -35,3 +31,13 @@ export function isDataQuery(url: string): boolean {
 export function isLocalUrl(url: string) {
   return !url.match(/^http/);
 }
+
+function _getRefId(num: number): string {
```
Comment (hugohaggmark) [inline public/app/core/utils/query.ts:35]: nit: please remove the underscore

File: public/app/core/utils/query.test.ts [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-    refId: 'C',
-  },
-  {
-    refId: 'D',
-  },
-  {
-    refId: 'E',
-  },
-];
+function _dataQueryHelper(ids: string[]): DataQuery[] {
```
Comment (hugohaggmark) [inline public/app/core/utils/query.test.ts:4]: nit: please remove the underscore

**Review by hugohaggmark — COMMENTED** [review]
We've discussed this internally, and we've come to the conclusion that this would be an ok addition. I left some very small nit comments.

**Review by hugohaggmark — APPROVED** [review]
Thank you for contributing, this looks great and works as expected 🎉

**Comment by adelvalle62** [comment]
Any update information when could be available the feature in version 8.1? I've got same issue when using SVG plugins like ACE SVG where is not possible to have more than 18 series or queries in a Panel, because the plugin seems to use internaly about 8 queries

**Comment by maximilian-weber** [comment]
is this in any release? can find any information in the release notes for 8.1.0

**Comment by hugohaggmark** [comment]
@adelvalle62 @maximilian-weber this was released in Grafana 8.1.0-beta1 but we forgot to add it to the change log. Sorry for the confusion.

**Comment by maximilian-weber** [comment]
@hugohaggmark thanks for the clarification. the documentations here could need an update: https://grafana.com/docs/grafana/latest/panels/queries/
i tried using the edit button, but i am getting a 404

**Comment by hugohaggmark** [comment]
Feel free to open a PR that corrects that documentation @maximilian-weber 🙏

**Comment by adelvalle62** [comment]
Anyway everything works fine since 8.1.0 in the monitoring tool. Thank you for all the support we received

Regards



De: Hugo Häggmark ***@***.***>
Responder a: grafana/grafana ***@***.***>
Fecha: jueves, 3 de febrero de 2022, 12:33 a. m.
Para: grafana/grafana ***@***.***>
CC: Adriano Del Valle ***@***.***>, Mention ***@***.***>
Asunto: Re: [grafana/grafana] Dashboard: Allow more than 26 queries per panel. (#35442)


@adelvalle62<https://github.com/adelvalle62> @maximilian-weber<https://github.com/maximilian-weber> this was released in Grafana 8.1.0-beta1 but we forgot to add it to the change log. Sorry for the confusion.

**Review by kelein — COMMENTED** [review]
<br>

**Pull request merged**